include/acacia/atspi/atspi_interface.h

```diff
@@ -0,0 +1,51 @@
+#ifndef INCLUDE_ACACIA_ATSPI_ATSPI_INTERFACE_H_
+#define INCLUDE_ACACIA_ATSPI_ATSPI_INTERFACE_H_
+
+#include <iostream>
+#include <memory>
+#include <string>
+
+#include <atspi/atspi.h>
+
+namespace acacia {
+
+/**
+ * Objects descending from this class wrap an
+ * [AtspiObject](https://docs.gtk.org/atspi2/class.Object.html) of a specific
+ * type. See descendants of this class for more information about each type.
+ */
+template <typename T>
+class AtspiInterface {
+ public:
+  explicit AtspiInterface(T* interface) : interface_(interface, Deleter()) {}
```

Comment:
`explicit` means it can't be used for implicit conversions and copy initialization: https://en.cppreference.com/w/cpp/language/explicit (the examples on that page are the clearest explanation.) 

That is, the constructor can't be used for assignment like `AtspiInterface<AtspiAction> i = pointer_to_atspi_action;`

I added it because I have had issues in this project with implicit conversions causing subtle bugs, particularly in SWIG-generated code. I wish I could remember what the issue was, but I remember it was a massive pain.

Reply:
I see, cool, ty!